C++/DataStructures/BST.cpp

```diff
@@ -0,0 +1,45 @@
+#include<iostream>
```

Comment:
Hi Anku!
One PR must only solve one issue. Your PR contains solutions for two different Issues.
Please use revert back to a single commit or create a separate branch for two different Issues. :smile: 

Here is a helpful link:
https://gist.github.com/hofmannsven/6814451